w3-revocations-check.md

```diff
@@ -0,0 +1,185 @@
+
+### Request Format
+
+#### URL Parameters
+
+**`cid`** (required): Delegation CID to check for revocations. Must be a valid IPFS CID string.
+
+#### Constraints
+
+- **CID Format:** Valid IPFS CID string (e.g., `bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi`)
+- **Encoding:** CID will be normalized to string format internally
+
+### Response Format
+
+#### Success Response (200): Delegation Revoked
+
+**Content-Type:** `application/vnd.ipld.car`
+
+Returns a CAR (Content Addressable aRchive) file containing:
+
+- Revocation metadata in `revocations@0.0.1` schema
+- Embedded UCAN proof blocks for trustless verification
+- ETag header for caching optimization
+
+##### CAR File Structure
+
+The CAR file contains a root block with the following structure:
+
+```json
+{
+  "revocations@0.0.1": {
+    "revocations": [
+      {
+        "delegation": { "/": "[delegationCID]" },
+        "scope": "[scopeDID]", 
+        "cause": { "/": "[causeCID]" }
+      }
+    ]
+  }
+}
+```
+
+##### Embedded Proof Blocks
+
+The CAR file includes the complete UCAN revocation proof referenced by `causeCID`, enabling clients to:
+
+- Cryptographically verify the revocation signature
+- Validate the revocation authority chain
+- Perform trustless verification without server dependency
+
+#### Not Found Response (404): Delegation Not Revoked
+
+**Content-Type:** `text/plain`
+
+```text
+Delegation not revoked
+```
+
+This indicates the delegation CID has not been explicitly revoked. Clients should check the delegation's proof chain to determine if it's invalid due to revoked dependencies.
+
+### Error Responses
+
+#### 400 Bad Request
+
+**Content-Type:** `application/json`
+
+##### Invalid CID Parameter
+
+```json
+{
+  "error": "Bad request",
+  "message": "Invalid CID parameter"
+}
+```
+
+##### Missing CID Parameter
+
+```json
+{
+  "error": "Bad request", 
+  "message": "CID parameter is required"
+}
+```
+
+#### 500 Internal Server Error
+
+**Content-Type:** `application/json`
+
+##### Query Failure
+
+```json
+{
+  "error": "Internal server error",
+  "message": "Failed to query revocations"
+}
+```
+
+##### General Error
+
+```json
+{
+  "error": "Internal server error",
+  "message": "An unexpected error occurred"
+}
+```
+
+### CORS Headers
+
+The endpoint includes CORS headers for cross-origin requests:
+
+- `Access-Control-Allow-Origin: *`
+- `Access-Control-Allow-Methods: GET`
+- `Access-Control-Allow-Headers: Accept`
+
+# Usage Examples
+
+## cURL Example
+
+```bash
+# Check if a delegation is revoked
+curl -X GET https://up.storacha.network/revocations/{cid}
+
+# Save CAR file for offline verification (when response is 200)
+curl -X GET https://up.storacha.network/revocations/{cid} \
+  -H "Accept: application/vnd.ipld.car" \
+  -o revocation-proof.car
+```
+
+## Client-Side Proof Chain Verification
+
+```javascript
+/**
+ * Recursively verify the delegation proof chain for revocations
```

Comment:
I would not...recurse. I'd collect the CIDs we want to check (start of chain first) and make parallel requests (with concurrency limit) allowing cancellation if one is found to be revoked.

I don't think it's worth putting this example in the spec - I'd probably remove.